core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -203,6 +215,16 @@ public void notifyNotLeader(Collection<TransactionContext> pendingEntries) {
     mJournalSystem.notifyLeadershipStateChanged(false);
   }
 
+  @Override
+  public CompletableFuture<TermIndex> notifyInstallSnapshotFromLeader(
+      RaftProtos.RoleInfoProto roleInfoProto, TermIndex firstTermIndexInLog) {
+    if (roleInfoProto.getRole() != RaftProtos.RaftPeerRole.FOLLOWER) {
+      return RaftJournalUtils.completeExceptionally(
+          new IllegalStateException("Received snapshot as a non-follower"));
```

Comment:
I'm not sure what this message means. Could you explain it or add more details in the message?

Reply:
Updated.